db/migrate/20161216223845_add_position_indices_to_pieces.rb

```diff
@@ -0,0 +1,5 @@
+class AddPositionIndicesToPieces < ActiveRecord::Migration
+  def change
+    add_index :pieces, [:y_position, :x_position], unique: true
```

Comment:
For this, you can't have a unique constraint. Because this is across the entire table.

So with this constraint, 2 different games wouldn't be able to have a piece in the same position.

If you want to do the unique constraint, this would work

`add_index :pieces, [:game_id, :y_position, :x_position], unique: true`